bakery/src/scripts/checksum-resources.py

```diff
@@ -0,0 +1,137 @@
+    md5s = []
+    try:
+        with open(filename, 'rb') as f:
+            while True:
+                data = f.read(BUF_SIZE)
+                if not data:
+                    break
+                sha1.update(data)
+                md5s.append(hashlib.md5(data))
+        # chunked calculation for AWS S3 MD5 etag
+        # https://stackoverflow.com/a/43819225/756056
+        if len(md5s) < 1:
+            s3_md5 = '"{}"'.format(hashlib.md5().hexdigest())
+        elif len(md5s) == 1:
+            s3_md5 = '"{}"'.format(md5s[0].hexdigest())
+        else:
+            digests = b''.join(m.digest() for m in md5s)
+            digests_md5 = hashlib.md5(digests)
+            s3_md5 = '"{}-{}"'.format(digests_md5.hexdigest(), len(md5s))
+        return sha1.hexdigest(), s3_md5
+    except IOError:     # file does not exist
+        return None, None
+
+
+def get_mime_type(filename):
+    """ get MIME type of file with libmagic """
+    mime_type = ''
+    try:
+        mime_type = magic.from_file(filename, mime=True)
+    finally:
+        return mime_type
+
+
+def create_symlink(img_filename, output_dir, sha1):
+    """ Create symlinks for resources to raw files """
+    raw_img_filename = os.path.realpath(img_filename)
+    dest_path = os.path.join(output_dir, RESOURCES_DIR)
+    dest_img_filename = os.path.join(dest_path, sha1)
+    rel_raw_img_filename = os.path.relpath(raw_img_filename, dest_path)
+    try:
+        # set a relative symlink to raw file
+        os.symlink(rel_raw_img_filename, dest_img_filename)
+    except FileExistsError:     # ignore symlink file existing
+        pass
+
+
+def create_json_metadata(output_dir, sha1, mime_type, s3_md5, original_name):
+    """ Create json with MIME type of a (symlinked) resource file """
+    data = {}
+    data['original_name'] = original_name
+    data['mime_type'] = mime_type
+    data['s3_md5'] = s3_md5
+    data['sha1'] = sha1
+    json_filename = os.path.join(output_dir, RESOURCES_DIR, sha1+'.json')
+    with open(json_filename, 'w') as outfile:
+        json.dump(data, outfile)
+
+
+def generate_checksum_resources_from_xhtml(filename, output_dir):
+    """ Go through HTML and checksum/copy/mime type all <img> and <a> resources """
+    doc = etree.parse(filename)
+    source_path = os.path.dirname(filename)
+    basename = os.path.basename(filename)
+    module_name = os.path.splitext(basename)[0]
+    # get all img @src resources
+    for node in doc.xpath('//x:img[@src and not(starts-with(@src, "http") or starts-with(@src, "//"))]',
+                          namespaces={'x': 'http://www.w3.org/1999/xhtml'}):
+        img_filename = node.attrib['src']
+        img_filename = os.path.join(source_path, img_filename)
+        img_basename = os.path.basename(img_filename)
+
+        sha1, s3_md5 = get_checksums(img_filename)
+        mime_type = get_mime_type(img_filename)
+
+        if sha1:    # file exists
+            node.attrib['src'] = '../' + RESOURCES_DIR + '/' + sha1
```

Comment:
```suggestion
            node.attrib['src'] = '/' + RESOURCES_DIR + '/' + sha1
```
Would an absolute path work just as well?

Reply:
Actually, I'm going to rescind this suggestion. Using a relative path allows it to work locally as well!